### ISSUE TITLE
Should not attempt to send subscription notifications when email not present

### ISSUE THREAD

**Issue by comp615** [rubysherpas/forem#650]
Right now when a user creates a post the system will attempt to send notifications to all previous posters (subscribers):
https://github.com/radar/forem/blob/rails4/app/models/forem/post.rb#L106

There is a bug here, however, in that at no point during the process does the system actually check if the user it's attempting to notify has an email (it's entirely plausible that they might not). This line should check to see if they have an email address that is valid before attempting to send:
https://github.com/radar/forem/blob/rails4/app/mailers/forem/subscription_mailer.rb#L10

**Comment by radar** [comment]
Hi @comp615,

It sure sounds like you know what the problem is and how to fix it. Could you please submit a PR that fixes this? Personally I would not put the check in the SubscriptionMailer, but on the method that calls the mailer.

**Comment by comp615** [comment]
Yeah that was my plan, this is sort of a marker for me as well :-P I'll
take a stab at it when I have a chance

**Comment by krainboltgreene** [comment]
@comp615 Any progress on this?
